app/forms/idv/api_image_upload_form.rb

```diff
@@ -21,19 +21,15 @@ def initialize(params, liveness_checking_enabled:, issuer:, analytics: nil)
 
     def submit
       throttled_else_increment
-      @form_response = validate_form
+      form_response = validate_form
 
       client_response = nil
       doc_pii_response = nil
 
       if form_response.success?
         client_response = post_images_to_client
 
-        if client_response.success?
-          doc_pii_response = validate_pii_from_doc(client_response)
-        else
-          client_response = client_response.merge(form_response)
-        end
+        doc_pii_response = validate_pii_from_doc(client_response) if client_response.success?
       end
 
       return determine_response(
```

Comment:
Do all responses include the remaining attempts extra data? Could we tack that on at the end in this `determine_response` logic?

Reply:
To do this in `determine_response`, there would either be merging on all current lines, e.g.
```
      return form_response.merge(extra_attributes_response) unless ...
      return doc_pii_response.merge(extra_attributes_response) if ...
      client_response.merge(extra_attributes_response)
```
or we'd have to jigger the logic to only have one return at the end, e.g.,
```
      response =  form_response unless ...
      response ||= doc_pii_response if ...
      response ||= client_response
      response.merge(extra_attributes_response)
```
I'm not fond of either of these, as I think they distract from the already confusing logic.